src/lib/UXDController.ts

```diff
@@ -0,0 +1,206 @@
+import { BigNumber, Contract, ContractTransaction, providers, Signer } from 'ethers';
+import { MintedEventObject, RedeemedEventObject, UXDControllerContract, UXDController as UXDControllerNamespace } from '../artifacts/types/UXDController';
+import { ERC20 as ERC20Contract, ApprovalEventObject, TransferEventObject } from '../artifacts/types/ERC20';
+import { Subject } from 'rxjs';
+import { ERC20__factory, UXDController__factory } from '../artifacts/types';
+
+export interface CollateralInfo {
+  symbol?: string;
+  token: string;
+  redeemable: BigNumber;
+  minted: BigNumber;
+}
+
+export class UXDController {
+  protected provider: providers.Provider;
+
+  // internal contracts
+  protected controllerContract: UXDControllerContract;
+  protected uxdContract: ERC20Contract;
+
+  // clients can listen to events on these subjects
+  public readonly mintSubject: Subject<MintedEventObject> = new Subject<MintedEventObject>();
+  public readonly redeemSubject: Subject<RedeemedEventObject> = new Subject<RedeemedEventObject>();
+  public readonly uxdApprovalSubject: Subject<ApprovalEventObject> = new Subject<ApprovalEventObject>();
+  public readonly uxdTransferSubject: Subject<TransferEventObject> = new Subject<TransferEventObject>();
+
+  constructor({
+    provider,
+    controllerAddress,
+    uxdTokenAddress,
+  }: {
+    provider: providers.Provider;
+    controllerAddress: string;
+    uxdTokenAddress: string;
+  }) {
+    this.provider = provider;
+
+    this.controllerContract = UXDController__factory.connect(controllerAddress, this.provider);
+    this.uxdContract = ERC20__factory.connect(uxdTokenAddress, this.provider);
+
+    this.registerEventListeners();
+  }
+
+  public mint({
+    market,
+    collateral,
+    ethAmount,
```

Comment:
`amount: number` would be nice, with a comment that it accepts a UI amount, and handles the BigNumber conversion if that's not too much trouble